### PR TITLE
Update vomit organ smite to not use Component.owner

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Title

## Why / Balance

Project 0 warnings, Component.Owner() is obsolete

## Technical details

Uses the EntityUid of the organs rather than the Owner of the xform component of the organs
Implement version of `GetBodyOrganComponents<T>` that returns the `Entity<T>` called `GetBodyOrganEntityComps`

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/3bd0f86f-2188-45c2-a035-a5e1c693fbb3

## Breaking changes

nah

**Changelog**

not player facing (for admins, people's organs now land inside containers and whatnot)

### CODE REVIEW

**Review by Tayrtahn — APPROVED** [review]

File: Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs [hunk 1]
```diff
@@ -285,18 +285,18 @@ private void AddSmiteVerbs(GetVerbsEvent<Verb> args)
             {
                 Text = "admin-smite-remove-hands-name",
                 Category = VerbCategory.Smite,
-                Icon = new SpriteSpecifier.Rsi(new ("/Textures/Fluids/vomit_toxin.rsi"), "vomit_toxin-1"),
+                Icon = new SpriteSpecifier.Rsi(new("/Textures/Fluids/vomit_toxin.rsi"), "vomit_toxin-1"),
                 Act = () =>
                 {
                     _vomitSystem.Vomit(args.Target, -1000, -1000); // You feel hollow!
-                    var organs = _bodySystem.GetBodyOrganComponents<TransformComponent>(args.Target, body);
```
Comment (deltanedas) [inline Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: this should just be changed to return `Entity<T>`

Comment (Plykiya) [reply Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: oh, can I do that?

Comment (Plykiya) [reply Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: ah, I can't just do that... I'm assuming you want me to make a version of GetBodyOrganComponents that just returns a List of Entity<T> instead

Comment (Plykiya) [reply Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: Is uh, whatever I did, what you wanted?

Comment (deltanedas) [reply Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: add OrganComponent to the Entity, it supports multiple components

Comment (Plykiya) [reply Content.Server/Administration/Systems/AdminVerbSystem.Smites.cs:292]: image
hmmmm

**Comment by Plykiya** [comment]
turns out I can use that new function in like four other PRs

File: Content.Shared/Body/Systems/SharedBodySystem.Organs.cs [hunk 2]
```diff
@@ -183,6 +183,32 @@ public bool RemoveOrgan(EntityUid organId, OrganComponent? organ = null)
...
+    /// <summary>
+    /// Returns a list of Entity<<see cref="T"/>, <see cref="OrganComponent"/>>
+    /// for each organ of the body
+    /// </summary>
+    /// <typeparam name="T">The component that we want to return</typeparam>
+    /// <param name="bodyUid">The EntityUid of the body to check the organs of</param>
+    /// <param name="body">The BodyComponent of the body to check the organs of</param>
+    public List<Entity<T, OrganComponent>> GetBodyOrganEntityComps<T>(
+        EntityUid bodyUid,
+        BodyComponent? body = null)
```
Comment (0x6273) [inline Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:195]: This should take `Entity<BodyComponent?>` instead.

Comment (Plykiya) [reply Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:195]: alright, sure

**Pull request merged**